.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,55 @@
+name: deploy
+
+on:
+  pull_request:
+  push:
+    branches: [main, '*-assessment-year']
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  publish-docker-image:
+    runs-on: ubuntu-latest
+    # Set token permissions so that we can publish to GitHub Container Registry
+    permissions:
+      contents: read
+      packages: write
```

Comment:
**question:** Are these strictly necessary? I haven't set them up in [the other GHA Docker builds](https://github.com/ccao-data/app-shiny-rpie/blob/master/.github/workflows/docker-build.yaml) and they seem to work fine.